teuthology/test/test_matrix.py

```diff
@@ -0,0 +1,70 @@
+from .. import matrix
+
+def do_test(res):
+    sz = res.size()
+    s = frozenset([matrix.generate_lists(res.index(i)) for i in range(sz)])
+    print sz, len(s)
+    if sz != len(s):
+        print sz
+    for i in range(res.size()):
+        print sorted([j for j in matrix.generate_lists(res.index(i))])
+        assert sz == len(s)
+
+def mbs(num, l):
+    return matrix.Sum(num*10, [matrix.Base(i + (100*num)) for i in l])
+
+class TestMatrix(object):
+    def test_simple(self):
```

Comment:
good naming conventions for unit tests is to have meaningful test methods, so that when they fail they can indicate what the failure is. 'test_matrix'  or 'test_simple' are obscure into what the intent of the test is.

This also allows for the test to become generic in the future because the signature allows it (since it is vague).

Also, like @andrewschoen mentioned, the `do_test` obscures further what the intent is here.